### DISCUSSION TITLE
gen_vx_mask as part of feature-relative use-case

### DISCUSSION THREAD

**Discussion by hertneky** [dtcenter/METplus#1308] [Configuration] [answered]
I am using the MTD feature-relative use-case for snow-bands over 6 cases. The process includes:
1) Masking the fcst and obs with ptype categorical snow (CSNOW) to identify the snowband objects
2) Running MTD feature-relative use-case on the masked data (runs MTD-extractTiles-SeriesAnalysis)

I thought it would be nice to enhance/expand upon this use-case by including the masking process as part of the use-case (i.e. as part of the process list in the conf), but I don't see a way to mask both datasets (fcst and obs) in a single METplus run. Is this currently possible or will the masking need to be run separately? As an example of the data I am using, for now the model/obs (st4) data are both masked using the model CSNOW variable. However, I plan to switch to using MRMS for obs, and mask using the precipFlag (3 for snow), so in that case the mask field would be different between the fcst and obs.

**Comment by j-opatz** [comment]
Hello,

And thank you for your question.

As you've discovered, masking will have to take place separately from the MTD tool call. However, you should be able to accomplish all of this in the same METplus configuration file. Just be sure to include two calls to the gen-vx-mask tool (one for each field you want to mask) before the MTD tool call and you should be able to only have to call METplus once. It should look like

PROCESS_LIST = GenVxMask, GenVxMask, MTD

I hope that answers your question. If not, please let us know what other question you might have or how we can better clarify the answer.

**Reply by hertneky** [reply]
That makes sense for masking two different fields for a single dataset; however, my question was regarding 2 different datasets. I need to mask both the fcst and obs datasets that are being input into MTD. Each are in 2 separate input directories, have different templates, etc

**Comment by georgemccabe** [comment, marked as answer]
Hi @hertneky,

You can set up 2 instances of GenVxMask that can process each data set. Example 3 from this section of the METplus User's Guide shows an example of this:

https://metplus.readthedocs.io/en/latest/Users_Guide/systemconfiguration.html#process-list

You will need to specify an instance name for one (or both) of the instances of GenVxMask, then create a section in the config file with the same name as the instance identifier to set values for that instance only. It would look something like this:

```
[config]
PROCESS_LIST = GenVxMask(fcst), GenVxMask(obs), MTD

...

[fcst]
GEN_VX_MASK_INPUT_TEMPLATE = /fcst/data/path

...

[obs]
GEN_VX_MASK_INPUT_TEMPLATE = /obs/data/path
```
